assembl/views/api2/auth.py

```diff
@@ -363,37 +363,24 @@ def reset_password(request):
     elif identifier:
         user, account = from_identifier(identifier)
         if not user:
-            if not discussion.preferences['generic_errors']:
-                raise JSONError(
-                    localizer.translate(_("This email does not exist")),
-                    code=HTTPNotFound.code)
-            else:
-                raise JSONError(
-                    localizer.translate(_(generic_error_message)),
-                    code=HTTPNotFound.code)
-                logger.error("This email does not exist.")
+            maybe_send_email(request, user, email, discussion=discussion)
         if account:
             email = account.email
     else:
         error = localizer.translate(_("Please give an identifier"))
         raise JSONError(error)
     if not email:
-        email = user.get_preferred_email()
+        if user:
+            email = user.get_preferred_email()
+        else:
+            email = None
         if not email:
-            if not discussion.preferences['generic_errors']:
-                error = localizer.translate(_("This user has no email"))
-            else:
-                error = localizer.translate(_(generic_error_message))
-                logger.error("This user has no email.")
-            raise JSONError(error, code=HTTPPreconditionFailed.code)
+            logger.error("This user has no email.")
+            maybe_send_email(request, user, email, discussion=discussion)
     if not isinstance(user, User):
-        if not discussion.preferences['generic_errors']:
-            error = localizer.translate(_("This is not a user"))
-        else:
-            error = localizer.translate(_(generic_error_message))
-            logger.error("This is not a user.")
-        raise JSONError(error, code=HTTPPreconditionFailed.code)
-    send_change_password_email(request, user, email, discussion=discussion)
+        logger.error("This is not a user.")
+        maybe_send_email(request, user, email, discussion=discussion)
```

Comment:
We do a check `not isinstance(user, User)` but we lose completely the reason why we shouldn't send the mail in maybe_send_email because we check only if profile is not None there. For example if it's just an AgentProfile instance and not a User instance.
Really I'm not sure about your `maybe_send_email` idea @ayyazdaniaryan 
I would just log the error here and `return HTTPOk()`

Reply:
I agree in this instance that the return HTTPOk() avoids the need to descend into the maybe_send_email